src/metabase/util/stats.clj

```diff
@@ -0,0 +1,315 @@
+  [question]
+    {:total 1
+     :native (if (= (question :iquery_type) "native") 1 0)
+     :gui (if (not= (question :iquery_type) "native") 1 0)})
+
+(defn- get-question-metrics
+  "Get metrics based on questions
+  TODO characterize by # executions and avg latency"
+  []
+  (let [questions (db/select 'Card)]
+    {:questions (apply add-summaries (map question-dims questions))}))
+
+(defn- get-dashboard-metrics
+  "Get metrics based on dashboards
+  TODO characterize by # of revisions, and created by an admin"
+  []
+  (let [dashboards (db/select 'Dashboard)
+        dashcards (db/select 'DashboardCard)]
+    {:dashboards (count dashboards)
+     :num_dashs_per_user (medium-histogram dashboards :creator_id)
+     :num_cards_per_dash (medium-histogram dashcards :dashboard_id)
+     :num_dashs_per_card (medium-histogram dashcards :card_id)}))
+
+(defn- get-pulse-metrics
+  "Get metrics based on pulses
+  TODO: characterize by non-user account emails, # emails"
+  []
+  (let [pulses (db/select 'Pulse)
+        pulsecards (db/select 'PulseCard)
+        pulsechannels (db/select 'PulseChannel)]
+    {:pulses (count pulses)
+     :pulse-types (frequencies (map :channel_type pulsechannels))
+     :pulse-schedules (frequencies (map :schedule_type pulsechannels))
+     :num_pulses_per_user (medium-histogram pulses :creator_id)
+     :num_pulses_per_card (medium-histogram pulsecards :card_id)
+     :num_cards_per_pulses (medium-histogram pulsecards :pulse_id)}))
+
+
+(defn- get-label-metrics
+  "Get metrics based on labels"
+  []
+  (let [labels (db/select 'CardLabel)]
+    {:labels (count labels)
+     :num_labels_per_card (micro-histogram labels :card_id)
+     :num_cards_per_label (medium-histogram labels :label_id)}))
+
+;; Metadata Metrics
+(defn- database-dims
+  "characterize a database record"
+  [database]
+  {:total 1
+   :analysed (if (database :is_full_sync) 1 0)})
+
+(defn- get-database-metrics
+  "Get metrics based on databases"
+  []
+  (let [databases (db/select 'Database)]
+    {:databases (apply add-summaries (map database-dims databases))}))
+
+
+(defn- get-table-metrics
+  "Get metrics based on tables"
+  []
+  (let [tables (db/select 'Table)]
+    {:tables (count tables)
+     :num_per_database (medium-histogram tables :db_id)
+     :num_per_schema (medium-histogram tables :schema)}))
+
+
+(defn- get-field-metrics
+  "Get metrics based on fields"
+  []
+  (let [fields (db/select 'Field)]
+    {:fields (count fields)
+     :num_per_table (medium-histogram fields :table_id)}))
+
+
+(defn- get-segment-metrics
+  "Get metrics based on segments"
+  []
+  (let [segments (db/select 'Segment)]
+    {:segments (count segments)}))
+
+
+(defn- get-metric-metrics
+  "Get metrics based on metrics"
+  []
+  (let [metrics (db/select 'Metric)]
+    {:metrics (count metrics)}))
+
+
+(defn- bin-latencies
+  "Bin latencies, which are in milliseconds"
+  [query-executions]
+  (let [latency-vals (map #(/ %1 1000) (map :running_time query-executions))]
```

Comment:
You should only number anonymous function literal args if there's more than one.

e.g. `#(/ % 1000)` instead of `#(/ %1 1000)`